### DISCUSSION TITLE
--debug命令可以正常推理，但是不使用debug则无法推理

### DISCUSSION THREAD

**Discussion by ZzzybEric** [open-compass/opencompass#429] [Q&A] [answered]
我们将本地模型部署在服务端，包装成openai接口，在用opencompass推理时，使用如下指令
python3 run.py configs/eval_gpt3.5.py -w outputs/demo --debug，此时接口能正常接收prompt

但是使用如下指令时python3 run.py configs/eval_gpt3.5.py -w outputs/demo，也就是去掉debug，outputs/demo/2023135631/logs/eval/GPT-3.5-turbo/ceval-computer_network.out里则显示prediction no found，接口也不会被调用

另外outputs/demo/2023135631/logs/infer/GPT-3.5-turbo/ceval-computer_network.out里只有一行内容/bin/sh: 1: python: not found

**Comment by gaotongxiao** [comment, marked as answer]
In non-debugging mode, tasks are performed in independent processes via commands starting with `python`. Apparently, your environment does not have `python`.
You may set up a global soft link in /usr/bin or somewhere else and link `python` to `python3`.

**Reply by ZzzybEric** [reply]
Thank you for your reply, it has solved my problem. But I would like to know why I used a local model for inference before, using Python 3 directly without debugging, and can still infer the results normally. 
"python3 run.py --models hf_opt_125m --datasets ceval_gen"
opt_125m is downloaded locally, and the model address in the configuration file is also changed to local
Please help me answer this question. Thank you

**Reply by gaotongxiao** [reply]
The differences come from here:

https://github.com/open-compass/opencompass/blob/681d3013de408166b6960dce5527627026da53ca/opencompass/tasks/openicl_infer.py#L45-L53

API model does not require GPU, and the execution command starts with "python". The commands used to run GPU-based models start with `torchrun`, and therefore bypass the error you encountered.

**Reply by ZzzybEric** [reply]
Thanks !
